### PR TITLE
Create assets

### PR DESCRIPTION
**Pull request by danielsteenberg**
Added Category database and Side navigation for category type with search function through params. Some styling done as well

### CODE REVIEW

File: app/assets/stylesheets/homepage.css.scss [hunk 1]
```diff
@@ -1,6 +1,17 @@
 .header {
```
Comment (jamieandyou) [inline app/assets/stylesheets/homepage.css.scss]: Remove .css from filename

File: app/assets/stylesheets/asset-nav-bar.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
+.asset-nav-bar li a {
```
Comment (jamieandyou) [inline app/assets/stylesheets/asset-nav-bar.scss]: Can be nesting the li components inside
 .asset-nav-bar {
     li {
That's the beauty of sass

File: app/assets/stylesheets/asset-nav-bar.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    color: white;
+}
+
+.asset-nav-bar ul {
+    list-style-type: none;
+    margin: 0;
+    padding: 0;
+    width: 25%;
+    background-color: #f1f1f1;
+    height: 100%; /* Full height */
```
Comment (jamieandyou) [inline app/assets/stylesheets/asset-nav-bar.scss]: Remove the comments

File: app/assets/stylesheets/asset-nav-bar.scss [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+.asset-nav {
+  width: 10%;
+  height: 100%;
+  position: relative;
+  top: 20px;
+  bottom: 0;
+  z-index: 1000;
+  background-color: #222;
+  margin-top: -40px;
```
Comment (jamieandyou) [inline app/assets/stylesheets/asset-nav-bar.scss]: See if you can avoid negative margins, It shows that you are compensating for something else. However, sometimes they are necessary. But try avoid them.

File: app/assets/stylesheets/asset-nav-bar.scss [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  margin-bottom: 100%;
+  padding-bottom: 100%;
+  float: left;
+}
+
+.label-asset {
+  color: white;
```
Comment (jamieandyou) [inline app/assets/stylesheets/asset-nav-bar.scss]: #fff

File: app/assets/stylesheets/asset-nav-bar.scss [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  font-size: 28px;
+}
+ul {
+  list-style-type: none;
```
Comment (jamieandyou) [inline app/assets/stylesheets/asset-nav-bar.scss]: You did this above for .asset-nav-bar ul

File: app/assets/stylesheets/employee-page.scss [hunk 7]
```diff
@@ -1,9 +1,9 @@
 strong {
-	font-weight: bold; 
+	font-weight: bold;
```
Comment (jamieandyou) [inline app/assets/stylesheets/employee-page.scss]: you shouldn't have to do this, theses are default styles

File: app/assets/stylesheets/homepage.css.scss [hunk 8]
```diff
@@ -1,6 +1,17 @@
...
-	width: 100%;
-	height: 15%;
-	background: black;
-	color: white; 
-}
\ No newline at end of file
+  width: 100%;
+  height: 15%;
+  background: black;
```
Comment (jamieandyou) [inline app/assets/stylesheets/homepage.css.scss]: #000

File: app/assets/stylesheets/homepage.css.scss [hunk 9]
```diff
@@ -1,6 +1,17 @@
...
+  color: white;
```
Comment (jamieandyou) [inline app/assets/stylesheets/homepage.css.scss]: #fff

File: app/assets/stylesheets/modern-business.css [hunk 10]
```diff
@@ -14,6 +14,12 @@ body {
 body {
     padding-top: 50px; /* Required padding for .navbar-fixed-top. Remove if using .navbar-static-top. Change if height of navigation changes. */
 }
+.container {
+    float: left;
+}
+.container table{
+    float: left;
+}
```
Comment (jamieandyou) [inline app/assets/stylesheets/modern-business.css]: Make this a scss file and nest the table

File: app/assets/stylesheets/searchbar.scss [hunk 11]
```diff
@@ -1,18 +1,19 @@
...
-	float: right;
-	border: 2px solid;
-	width: 20%;
-	overflow: hidden;
-	box-shadow: 3px 3px 2px #888888;
+  float: right;
+  border: 2px solid;
+  width: 20%;
+  overflow: hidden;
+  box-shadow: 3px 3px 2px #888888;
```
Comment (jamieandyou) [inline app/assets/stylesheets/searchbar.scss]: #888

File: app/models/asset.rb [hunk 12]
```diff
@@ -1,3 +1,4 @@
 class Asset < ActiveRecord::Base
   belongs_to :employee
+  has_many :categories
```
Comment (jamieandyou) [inline app/models/asset.rb]: Isn't it has_one?

File: app/views/assets/_form.html.slim [hunk 13]
```diff
@@ -23,6 +23,7 @@
       = f.text_field :assignee
     .field
       = f.label :category
-      = f.text_field :category
+      br /
```
Comment (jamieandyou) [inline app/views/assets/_form.html.slim]: Remove the /

File: app/views/shared/_category_table.html.slim [hunk 14]
```diff
@@ -0,0 +1,5 @@
+table
+tbody
+  @categories.each do |category|
+  tr
+    td = link_to category.category, laptops_path, class: 'btn btn-primary'
```
Comment (jamieandyou) [inline app/views/shared/_category_table.html.slim]: I don't think this is supposed to be laptops_path

File: config/routes.rb [hunk 15]
```diff
@@ -2,9 +2,10 @@
 
   scope path: "asset-directory" do
     resources :assets, as: 'assets', path: 'all'
-    get 'laptops', to: 'assets#laptops', as: 'laptops', path: 'laptops'
+    get 'category_list', to: 'assets#category_list', as: 'category-list', path: 'category-list'
```
Comment (jamieandyou) [inline config/routes.rb]: localhost:3000/asset-directory/items?category=mice

File: app/assets/stylesheets/homepage.css.scss (hunk 1 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/homepage.css.scss]: done

File: app/assets/stylesheets/asset-nav-bar.scss (hunk 3 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/asset-nav-bar.scss]: Done

File: app/assets/stylesheets/asset-nav-bar.scss (hunk 5 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/asset-nav-bar.scss]: Done

File: app/assets/stylesheets/asset-nav-bar.scss (hunk 6 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/asset-nav-bar.scss]: Done

File: app/assets/stylesheets/employee-page.scss (hunk 7 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/employee-page.scss]: removed

File: app/assets/stylesheets/homepage.css.scss (hunk 8 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/homepage.css.scss]: Changed

File: app/assets/stylesheets/homepage.css.scss (hunk 9 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/homepage.css.scss]: Changed

File: app/assets/stylesheets/searchbar.scss (hunk 11 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/searchbar.scss]: Changed

File: app/models/asset.rb (hunk 12 above)
Comment (danielsteenberg) [inline app/models/asset.rb]: Changed, an asset belongs to one category is correct

File: app/views/assets/_form.html.slim (hunk 13 above)
Comment (danielsteenberg) [inline app/views/assets/_form.html.slim]: removed

File: app/views/shared/_category_table.html.slim (hunk 14 above)
Comment (danielsteenberg) [inline app/views/shared/_category_table.html.slim]: Deleted the whole page

File: app/assets/stylesheets/asset-nav-bar.scss (hunk 2 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/asset-nav-bar.scss]: Sorted

File: app/assets/stylesheets/modern-business.css (hunk 10 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/modern-business.css]: Sorted

File: config/routes.rb (hunk 15 above)
Comment (danielsteenberg) [inline config/routes.rb]: sorted

File: app/assets/stylesheets/employee-page.scss [hunk 16]
```diff
@@ -1,7 +1,3 @@
-strong {
-	font-weight: bold;
-}
-
 em {
 	font-style: italic;
 }
```
Comment (jamieandyou) [inline app/assets/stylesheets/employee-page.scss]: This is also a default style

File: app/models/asset.rb [hunk 17]
```diff
@@ -1,3 +1,4 @@
 class Asset < ActiveRecord::Base
   belongs_to :employee
+  has_one :categories
```
Comment (jamieandyou) [inline app/models/asset.rb]: This should maybe be singular :category

File: app/assets/stylesheets/employee-page.scss [hunk 18]
```diff
@@ -1,9 +1,5 @@
-strong {
-	font-weight: bold; 
-}
-
 em {
```
Comment (jamieandyou) [inline app/assets/stylesheets/employee-page.scss]: The indentation on this whole file must be reduced

File: app/assets/stylesheets/asset-nav-bar.scss [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+    list-style-type: none;
+    margin: 0;
+    padding: 0;
+    width: 25%;
+    background-color: #f1f1f1;
+    height: 100%;
+    position: fixed;
+    overflow: auto;
+  }
+
```
Comment (jamieandyou) [inline app/assets/stylesheets/asset-nav-bar.scss]: remove this space line

File: app/assets/stylesheets/employee-page.scss (hunk 16 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/employee-page.scss]: Deleted

File: app/models/asset.rb (hunk 17 above)
Comment (danielsteenberg) [inline app/models/asset.rb]: changed to singular

File: app/assets/stylesheets/employee-page.scss (hunk 18 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/employee-page.scss]: The file is two line spaces but set the tabs to two line spaces and hope that is fixed for github

File: app/assets/stylesheets/asset-nav-bar.scss (hunk 19 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/asset-nav-bar.scss]: removed spaced line

File: app/assets/stylesheets/asset-nav-bar.scss (hunk 4 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/asset-nav-bar.scss]: Deleted the negative Margin, found why i did it and solved the issue

File: app/models/asset.rb (hunk 17 above)
Comment (danielsteenberg) [inline app/models/asset.rb]: When making it singular it breaks the program, saying that it has an association issue.

File: app/assets/stylesheets/asset-nav-bar.scss [hunk 20]
```diff
@@ -0,0 +1,49 @@
+.asset-nav-bar {
+  li a{
```
Comment (jamieandyou) [inline app/assets/stylesheets/asset-nav-bar.scss]: You need to be consistent, so make it `li a {` with a space

File: app/assets/stylesheets/asset-nav-bar.scss [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+    display: block;
+    color: #fff;
+    padding: 8px 0 8px 16px;
```
Comment (jamieandyou) [inline app/assets/stylesheets/asset-nav-bar.scss]: Put px after each one

File: app/assets/stylesheets/nav.scss [hunk 22]
```diff
@@ -35,11 +35,15 @@ title {
...
+  border-width: 0 0 1px;
+  position: fixed;
+}
+.navbar-inverse {
+  background-color: #222;
+  border-color: #222;
 }
 
 .navbar-right {
   color: blue;
```
Comment (jamieandyou) [inline app/assets/stylesheets/nav.scss]: This must be a hex code

File: app/views/assets/_form.html.slim [hunk 23]
```diff
@@ -22,7 +22,8 @@
       = f.label :assignee
       = f.text_field :assignee
     .field
-      = f.label :category
-      = f.text_field :category
+      = f.label :category_id
```
Comment (jamieandyou) [inline app/views/assets/_form.html.slim]: Is this label going to say "Category ID"?

If it is, maybe just write out the exact label that you want e.g. "Category" in normal HTML

File: app/views/assets/show.html.slim [hunk 24]
```diff
@@ -21,7 +21,7 @@
       = @asset.assignee
     p
       strong Item Category:
-      = @asset.category
+      = @asset.category_id
```
Comment (jamieandyou) [inline app/views/assets/show.html.slim]: Surely you want to show the category name here

File: app/views/categories/_form.html.slim [hunk 25]
```diff
@@ -0,0 +1,13 @@
+= form_for @category, html: { multipart: true , class: 'form-horizontal' } do |f|
+  - if @category.errors.any?
+    #error_explanation
+      h2 = "#{pluralize(@category.errors.count, "error")} prohibited this asset from being saved:"
+      ul
+        - @category.errors.full_messages.each do |message|
+          li = message
+  .form
+    .field
+      = f.label :category
```
Comment (jamieandyou) [inline app/views/categories/_form.html.slim]: Label should be :name

File: app/assets/stylesheets/asset-nav-bar.scss (hunk 20 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/asset-nav-bar.scss]: Done

File: app/assets/stylesheets/asset-nav-bar.scss (hunk 21 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/asset-nav-bar.scss]: Done

File: app/assets/stylesheets/nav.scss (hunk 22 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/nav.scss]: Done

File: app/views/assets/_form.html.slim (hunk 23 above)
Comment (danielsteenberg) [inline app/views/assets/_form.html.slim]: Changed it back to category

File: app/views/assets/show.html.slim (hunk 24 above)
Comment (danielsteenberg) [inline app/views/assets/show.html.slim]: changed it to name

File: app/views/categories/_form.html.slim (hunk 25 above)
Comment (danielsteenberg) [inline app/views/categories/_form.html.slim]: done

File: app/assets/stylesheets/asset-nav-bar.scss [hunk 26]
```diff
@@ -27,8 +27,8 @@
   height: 100%;
   position: relative;
   top: 20px;
-  bottom: 0;
-  z-index: 1000;
+  bottom: 0px;
+  z-index: 1000px;
```
Comment (jamieandyou) [inline app/assets/stylesheets/asset-nav-bar.scss]: z-index doesn't have px at the end of it

File: app/assets/stylesheets/nav.scss [hunk 27]
```diff
@@ -3,26 +3,26 @@
 }
 
 .capture a:visited {
-  color: #FFFFFF;
+  color: #FFF;
```
Comment (jamieandyou) [inline app/assets/stylesheets/nav.scss]: Make all lower cases

File: app/assets/stylesheets/nav.scss [hunk 28]
```diff
@@ -45,5 +45,5 @@ title {
 }
 
 .navbar-right {
-  color: blue;
+  color: #0000FF;
```
Comment (jamieandyou) [inline app/assets/stylesheets/nav.scss]: lower case

File: app/assets/stylesheets/asset-nav-bar.scss (hunk 26 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/asset-nav-bar.scss]: sorted

File: app/assets/stylesheets/nav.scss (hunk 27 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/nav.scss]: sorted

File: app/assets/stylesheets/nav.scss (hunk 28 above)
Comment (danielsteenberg) [inline app/assets/stylesheets/nav.scss]: sorted

**Comment by jamieandyou** [comment]
👍

**Pull request merged**